packages/rest/src/controllers/didcomm/basic-messages/BasicMessagesController.ts

```diff
@@ -0,0 +1,72 @@
+import { RecordNotFoundError, BasicMessageRole } from '@credo-ts/core'
+import { Body, Controller, Example, Get, Post, Query, Request, Route, Security, Tags } from 'tsoa'
+import { injectable } from 'tsyringe'
+
+import { RequestWithAgent } from '../../../authentication'
+import { apiErrorResponse } from '../../../utils/response'
+import { RecordId, ThreadId } from '../../types'
+
+import { basicMessageRecordExample } from './BasicMessagesControllerExamples'
+import {
+  basicMessageRecordToApiModel,
+  DidCommBasicMessagesSendOptions,
+  type DidCommBasicMessagesRecord,
+} from './BasicMessagesControllerTypes'
+
+@Tags('DIDComm Basic Messages')
+@Route('/didcomm/basic-messages')
+@Security('tenants', ['tenant'])
+@injectable()
+export class DidCommBasicMessagesController extends Controller {
+  /**
+   * Retrieve basic messages by connection id
+   *
+   * @param connectionId Connection identifier
+   * @returns BasicMessageRecord[]
+   */
+  @Example<DidCommBasicMessagesRecord[]>([basicMessageRecordExample])
+  @Get('/')
+  public async findBasicMessagesByQuery(
+    @Request() request: RequestWithAgent,
+    @Query('connectionId') connectionId?: RecordId,
+    @Query('role') role?: BasicMessageRole,
+    @Query('threadId') threadId?: ThreadId,
+    @Query('parentThreadId') parentThreadId?: ThreadId,
+  ): Promise<DidCommBasicMessagesRecord[]> {
+    const basicMessageRecords = await request.user.agent.basicMessages.findAllByQuery({
+      connectionId,
+      role,
+      threadId,
+      parentThreadId,
+    })
+    return basicMessageRecords.map(basicMessageRecordToApiModel)
+  }
+
+  /**
+   * Send a basic message to a connection
+   *
+   * @param connectionId Connection identifier
+   * @param content The content of the message
+   * @returns BasicMessageRecord
+   */
+  @Example<DidCommBasicMessagesRecord>(basicMessageRecordExample)
+  @Post('/')
```

Comment:
Time will tell, but maybe a `post` on root can be a bit unclear that it will actually send a message. A `/send` endpoint (or `send-message` to align with module API) might be more clear.

I am fine with either, just worth writing a note about.

Reply:
Yes makes sense. There's quite some endpoints I think that we need to change in this case